### PR TITLE
ADBDEV-1745. Add an unconditional call to query_info_collect_hook in ExecProcNode

### PR DESCRIPTION
**Pull request by leskin-in**
Implement two unconditional calls to `query_info_collect_hook()`: in `ExecProcNode()` and in `MultiExecProcNode()`.

These calls enable extensions to gain control flow _during_ node execution. This, in turn, allows extensions to inspect query's runtime structures during query execution.

### CODE REVIEW

**Review by InnerLife0 — DISMISSED** [review]

File: src/backend/executor/execProcnode.c [hunk 1]
```diff
@@ -1242,6 +1245,9 @@ MultiExecProcNode(PlanState *node)
 	}
 
 	TRACE_POSTGRESQL_EXECPROCNODE_EXIT(GpIdentity.segindex, currentSliceId, nodeTag(node), node->plan->plan_node_id);
+
+	if (query_info_collect_hook)
```
Comment (darthunix) [inline src/backend/executor/execProcnode.c:1247]: I believe that `query_info_collect_hook` should be **always** inserted before `TRACE_POSTGRESQL_EXECPROCNODE_EXIT` macros. Though this macros is defined as empty, some debug extensions can redefine it and expect to be the last operation with the node (for time tracking for example).

Comment (darthunix) [reply src/backend/executor/execProcnode.c:1247]: The same thing is actual for the master branch [draft](https://github.com/arenadata/gpdb/pull/207).

Comment (leskin-in) [reply src/backend/executor/execProcnode.c:1247]: For exactly the same purpose (to obtain "last operation" guarantee) we may wish to leave the hook in its current location. The macro cannot be redefined at runtime, thus its user would have to be aware of the fact that it is not the last operation in `ExecProcNode()`.

Taking into account that `gpperfmon` packets [are sent earlier](https://github.com/arenadata/gpdb/pull/208/files#diff-1255433ef357570f11a775d6a32fab02584024c0708986e0b16837717ee355b0R1161) than the macro is called, the change you propose makes sense.

Comment (leskin-in) [reply src/backend/executor/execProcnode.c:1247]: Changed in https://github.com/arenadata/gpdb/pull/208/commits/7b7ceb0b5927b4eefe2fe37e6a80e3ddab5ff898.

**Review by darthunix — DISMISSED** [review]

File: src/backend/executor/execProcnode.c (hunk 1 above)
Comment (maksm90) [reply src/backend/executor/execProcnode.c:1247]: > some debug extensions can redefine it

not debug extensions but static tracing tools (ebpf tools, as example)

**Review by maksm90 — DISMISSED** [review]

**Review by darthunix — APPROVED** [review]

**Review by InnerLife0 — APPROVED** [review]

**Comment by leskin-in** [comment]
I wonder whether this PR will finally be merged 🙂

**Comment by Stolb27** [comment]
In the end, it was decided to rely on the gpdb infrastructure to poll instrumentation slots in the shared memory. I'm sorry(

**Pull request closed without merging**